src/main/java/org/jabref/gui/openoffice/DetectOpenOfficeInstallation.java

```diff
@@ -44,28 +51,27 @@ public DetectOpenOfficeInstallation(JDialog parent, OpenOfficePreferences prefer
     }
 
     public boolean isInstalled() {
-        foundPaths = false;
         if (this.checkAutoDetectedPaths(preferences)) {
             return true;
         }
         init();
-        getWorker().run();
-        update();
-        return foundPaths;
-    }
-
-    @Override
-    public void run() {
-        foundPaths = autoDetectPaths();
+        try {
+            Object result = BackgroundTask.wrap(this::autoDetectPaths)
+                                     .onSuccess(x -> SwingUtilities.invokeLater(progressDialog::dispose))
+                                     .executeWith(Globals.TASK_EXECUTOR)
+                                     .get();
```

Comment:
This looks a bit strange to me. The `get` makes the thread wait until the task is completely, so the operation is essentially synchronous. Its probably better to return the `Task<Boolean>` and let the caller decide if he wants to wait.